flash/data/process.py

```diff
@@ -295,17 +317,35 @@ def __init__(
         self._predict_collate_in_worker_from_transform: Optional[bool] = None
         self._test_collate_in_worker_from_transform: Optional[bool] = None
 
-        self.train_transform = convert_to_modules(self._check_transforms(train_transform, RunningStage.TRAINING))
-        self.val_transform = convert_to_modules(self._check_transforms(val_transform, RunningStage.VALIDATING))
-        self.test_transform = convert_to_modules(self._check_transforms(test_transform, RunningStage.TESTING))
-        self.predict_transform = convert_to_modules(self._check_transforms(predict_transform, RunningStage.PREDICTING))
+        # store the transform before conversion to modules.
+        self._train_transform = self._check_transforms(train_transform, RunningStage.TRAINING)
+        self._val_transform = self._check_transforms(val_transform, RunningStage.VALIDATING)
+        self._test_transform = self._check_transforms(test_transform, RunningStage.TESTING)
+        self._predict_transform = self._check_transforms(predict_transform, RunningStage.PREDICTING)
 
-        if not hasattr(self, "_skip_mutual_check"):
-            self._skip_mutual_check = False
+        self.train_transform = convert_to_modules(self._train_transform)
+        self.val_transform = convert_to_modules(self._val_transform)
+        self.test_transform = convert_to_modules(self._test_transform)
+        self.predict_transform = convert_to_modules(self._predict_transform)
 
         self._callbacks: List[FlashCallback] = []
 
-    # todo (tchaton) Add a warning if a transform is provided, but the hook hasn't been overriden !
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
+        preprocess_state_dict = self.get_state_dict()
+        if not isinstance(preprocess_state_dict, Dict):
+            raise MisconfigurationException("get_state_dict should return a dictionary")
+        preprocess_state_dict["_meta"] = {}
+        preprocess_state_dict["_meta"]["module"] = self.__module__
+        preprocess_state_dict["_meta"]["class_name"] = self.__class__.__name__
+        preprocess_state_dict["_meta"]["_state"] = self._state
+        try:
+            commit_sha = subprocess.check_output(["git", "rev-parse", "HEAD"]).decode("ascii").strip()
+        except subprocess.CalledProcessError:
+            commit_sha = "n/a"
+        preprocess_state_dict["_meta"]["commit_sha"] = commit_sha
```

Comment:
This is never used, do we still need it?

Reply:
Removed !